src/middleware/jira-symmetric-jwt-middleware.ts

```diff
@@ -0,0 +1,130 @@
+import { decodeSymmetric, getAlgorithm } from "atlassian-jwt";
+import Logger from "bunyan";
+import { NextFunction, Request, Response } from "express";
+import { booleanFlag, BooleanFlags } from "~/src/config/feature-flags";
+import { TokenType, verifyQsh } from "~/src/jira/util/jwt";
+import { Installation } from "~/src/models/installation";
+import { moduleUrls } from "~/src/routes/jira/atlassian-connect/jira-atlassian-connect-get";
+import { matchRouteWithPattern } from "~/src/util/match-route-with-pattern";
+
+export const jiraSymmetricJwtMiddleware = async (req: Request, res: Response, next: NextFunction) => {
+
+	if (!await booleanFlag(BooleanFlags.NEW_JWT_VALIDATION, false)) {
+		return next();
+	}
+
+	const token = req.query?.["jwt"] || req.cookies?.["jwt"];
+
+	if (token) {
+		const issuer = getIssuer(token, req.log);
+		if (!issuer) {
+			req.log.warn("JWT claim did not contain the issuer (iss) claim");
+			return res.status(401).send("Unauthorised");
+		}
+
+		const installation = await Installation.getForClientKey(issuer);
+		if (!installation) {
+			req.log.warn("No Installation found");
+			return res.status(401).send("Unauthorised");
+		}
+
+		const secret = await installation.decrypt("encryptedSharedSecret");
+
+		const tokenType = checkPathValidity(req.originalUrl) ? TokenType.normal : TokenType.context;
```

Comment:
This middleware only deals with Symmertric tokens not Asymmertric, so there is no need to determine token type.
Now Symmetric token could be with fixed or computed qsh.  That is decided using `checkPathValidity` method.

Reply:
`Symmetric token could be with fixed or computed qsh` <-- this is the `TokenType` I meant, `normal` or `context`

So I think the determine it is `fixed` or `computed qsh`, we should just look into the token itself, instead of depends on url.

---

However, to determine whether `computed qsh` or `fixed` is allowed, we should look into url. This is a separate thing compare to above.

----

But all goods, since this is the existing logics, we can leave this as it is.

Reply:
/nit
maybe rename `tokenType` to `expectedTokenType` or `allowedTokenType`? So that we know this is NOT the type of the token itself, but the type we are expecting (from url?)

Reply:
I think this will create ambigituy, leaving as it is for now.